bloom/generators/rpm/templates/ament_cmake/template.spec.em

```diff
@@ -56,6 +57,18 @@ mkdir -p obj-%{_target_platform} && cd obj-%{_target_platform}
 if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.sh"; fi
 %make_install -C obj-%{_target_platform}
 
+%if 0%{?with_tests}
+%check
+TEST_TARGET=$(%__make -qp -C obj-%{_target_platform} | sed "s/^\(test\|check\):.*/\\1/;t f;d;:f;q0")
```

Comment:
Can you elaborate what / why this is doing  checking for?

Same for the other build types.

Reply:
This is looking for any targets in the Makefile named either `test` or `check`, and saving the first one it encounters to an environment variable. If there are none, then the variable is empty.

Same logic for plain cmake.

For `ament_python`, the logic looks for a directory called either `test` or `tests`, and invokes pytest on the first one it finds.